x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -324,3 +318,65 @@ func TestCalcSingleAssetInAndOut_InverseRelationship(t *testing.T) {
 		}
 	}
 }
+
+func TestCalcJoinPoolShares(t *testing.T) {
+	testCases := []struct {
+		name         string
+		swapFee      sdk.Dec
+		poolAssets   []balancer.PoolAsset
+		tokensIn     sdk.Coins
+		expectErr    bool
+		expectShares sdk.Int
+		expectLiq    sdk.Coins
+	}{
+		{
+			name:    "equal weights with zero swap fee",
+			swapFee: sdk.MustNewDecFromStr("0"),
+			poolAssets: []balancer.PoolAsset{
+				{
+					Token:  sdk.NewInt64Coin("uosmo", 1_000_000_000_000),
+					Weight: sdk.NewInt(100),
+				},
+				{
+					Token:  sdk.NewInt64Coin("uatom", 1_000_000_000_000),
+					Weight: sdk.NewInt(100),
+				},
+			},
+			tokensIn:     sdk.NewCoins(sdk.NewInt64Coin("uosmo", 50_000)),
+			expectErr:    false,
+			expectShares: sdk.NewInt(2499999968800),
```

Comment:
How was this calculated @alexanderbez?

Reply:
😎😎

Reply:
By calling the function @AlpinYukseloglu 😆

Reply:
I'm down to have this merged as is, and handle more verification of these numbers in Alpin's follow-up PR